manifests/config.pp

```diff
@@ -19,6 +19,15 @@
     content => template('foreman/database.yml.erb'),
   }
 
+  if $foreman::email_delivery_method {
```

Comment:
One problem may be that if a user changes the method to be unmanaged, you have a left over file. Also puppet 4 is a bit different about empty strings so maybe use ```if $foreman::email_delivery_method && !empty($foreman::email_delivery_method)```?